org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm

```diff
@@ -70,159 +71,117 @@ class State(ABC):
-                return False
-            if self.isAbstract and not len(subset) == len(value):
-                logging.log(SVLOG, self.name + " does not conform to the abstract constraint.")
-                return False
-        if self.isSubsig:
-            siblings = [v for subsig in self.parent.subsigs - {self} for v in subsig.get_value()]
-            if len(value.intersection(siblings)) != 0:
-                logging.log(SVLOG, self.name + " is not a disjoint subset of " + self.parent.name + ".")
-                return False
-            if not self.parent.check_validity(self.parent.get_value()):
-                logging.log(SVLOG, self.name + "'s parent " + self.parent.name + " does not have valid value.")
-                return False
-
-        # Multiplicity
-        if (self.upper and len(value) > self.upper) or self.lower > len(value):
-            logging.log(SVLOG, self.name + " violates the multiplicity rule.")
-            return False
+    def __repr__(self):
+        return f"{self.sigName} - {self.name} : {self.elements}" if self.elements else f"{self.sigName} - {self.name} : Empty"
 
-        return True
+    def __contains__(self, other):
+        if isinstance(other, MetaSignature):
+            return other.possible_objects().issubset(self.elements)
+        if isinstance(other, Signature):
+            return other.elements.issubset(self.elements)
+        if isinstance(other, str):
+            return other in self.elements
+        return False
 
-    def get_value(self):
-        if self.subsigs:
-            subset = set([v for subsig in self.subsigs for v in subsig.get_value()])
-            return subset.union(self._value)
-        return self._value
-
-    # Returns None if the value intended to be set is invalid
-    def set_value(self, value):
-        if self.check_validity(value):
-            subset = set([v for subsig in self.subsigs for v in subsig.get_value()])
-            self._value = value - subset
-            return self._value
-        else:
-            return None
-
-    def set_value_no_constraints(self, value):
-        subset = set([v for subsig in self.subsigs for v in subsig.get_value()])
-        self._value = value - subset
-        return self._value
 
-    def __repr__(self):
-        return f"{self.name} {self.get_value()}"
+    def set_failed(self, new_elements) -> bool:
+        if(type(self).is_valid(new_elements)):
+            self.elements = new_elements
+            return False
+        return True
+
 
 # --- Signatures ---
 
-#foreach($signature in $signatures)
-${signature.Name} = Signature("${signature.Name}", "${signature.Multiplicity}", ${signature.Cardinality}, ${signature.IsSubset}, ${signature.Parents}, ${signature.IsSubsig}, ${signature.Parent}, ${signature.IsAbstract})
-#end
+#foreach($sig in $signatures)
+    #if(${sig.isSubset()})
+class ${sig.Name}(${sig.getParentsName()}):
```

Comment:
This multiple inheritance is not the same as a subset relationship. If we have
classes C1, C2, C3, then class C(C1, C2, C3) would be a subclass of C1 **and** C2 **and** C3.
Whereas for a Dash signature subset relationship C in C1+C2+C3, C can be one of C1, C2,C3.
I couldn't think of a good way to do this using Python built-in class inheritance, one possible approach could be take those atoms and make a new class whose instances would be a subset of those possible atoms.

Reply:
In fact, the inheritances right now are not used, this can actually be deleted. I.e., all signatures can be derived from Signature only. Subsigs too

Reply:
See my comment below.